### ISSUE TITLE
Queries don't work on indexed properties when property name is a different case than the index.

### ISSUE THREAD

**Issue by TT-Dev1** [orientechnologies/orientdb#4331]
- **Queries don't work on indexed properties when property name is a different case than the index.**
- Mixing cases with index fields makes unusable database

*It _IS_ possible that I created a strange case because the CLASS does not have the PROPERTY explicitly defined when importing via OETL -- i.e. Schemaless at that point (something I never usually do but a result of testing ETL import)*

A more detailed discussion about how this was discovered can be viewed here...  https://groups.google.com/forum/#!topic/orient-database/kV3JlasaD6I

Repro:

**1:**   I load a MSSQL table via ETL and I didn't previously create the class...   (i.e. I let OETL create the class via the importer.)

```
{
  "config": {
    "verbose": false,
    "log": "info"
  },
  "extractor" : {
    "jdbc": { "driver": "com.microsoft.sqlserver.jdbc.SQLServerDriver",
              "url": "jdbc:sqlserver://localhost:1433;databaseName=SomeDB",
              "userName": "sa",
              "userPassword": "xxxx",
              "query": "SELECT * FROM Players" }
  },
  "transformers" : [ { "field": { "fieldName": "rowguid", "operation": "remove"} },
    { "vertex": { "class": "Player", "skipDuplicates": true } }
  ],
  "loader" : {
    "orientdb": {
      "dbURL": "plocal:D:/orientdb/databases/MyDB",
      "dbUser": "admin",
      "dbPassword": "admin",
      "dbAutoCreate": true,
      dbAutoDropIfExists: false,
      tx: false,
      wal: false,
      batchCommit: 0,
      dbType: "graph",
      **indexes: [{"class":"Player", "fields":["PlayerID:integer"], "type":"UNIQUE" }]**
    }
  }
}
```

**2:**   I try to select all vertices with a certain id (INTEGER)

>    SELECT PlayerId, PlayerName FROM Player WHERE **PlayerId = 99**

Expected: get back my single VERTEX
Actual: I get back nothing

**Now the weird part....**

**3:   I select with a range and get back the record**

>    SELECT PlayerId, PlayerName FROM Player WHERE **PlayerId >= 99 AND PlayerID <= 99**

And, because a picture is worth a thousand words...

![orientdb-select1](https://cloud.githubusercontent.com/assets/10066158/8055276/fc62a5f0-0ea6-11e5-9329-08f7e54759d7.jpg)

And here is the schema

![orientdb-select2](https://cloud.githubusercontent.com/assets/10066158/8055316/57a80040-0ea7-11e5-8652-3f35cc2cbd8c.jpg)

** THE INTERESTING THING (and likely why the bug appears) is that the PROPERTY here is the same as the index and with a capitalized 'ID' while the imported value (seen previously has a mixed case 'Id'.) **

**The work-around?**

With the ETL using the mixed case index name that exactly matched the imported field name then everything works.

In ETL, changed...

> indexes: [{"class":"Player", "fields":["**PlayerID**:integer"], "type":"UNIQUE" }]

to

> indexes: [{"class":"Player", "fields":["**PlayerId**:integer"], "type":"UNIQUE" }]

and now everything works, with any case.

```
SELECT FROM Player WHERE PlayerId = 99
SELECT FROM player WHERE playerid = 99
SELECT FROM PLAYER WHERE PLAYERID = 99
SELECT FROM PlAyER WHERE pLaYeRiD = 99
```

*So, it appears that the problem is that something is getting out of sync for when the engine decides to use the index and when it doesn't.  Using a range precludes the index so that works without case issues but '=' does not.  Don't think it's necessarily ETL related but it seems like a bug to me regardless.*

**Comment by lvca** [comment]
I think ETL created PlayerID property in schema and index that rely on it. Could you check in the schema if you had both PlayerId and PlayerID ?

**Comment by TifosiTech** [comment]
Yes, I believe you are correct.  The schema (in the picture above) does not show all the properties because it is schemaless as the properties were created by the ETL process.

One fix could be to check existing schema (or data) and use the case of the imported data and not the case of the supplied "indexes" field.

Another fix could be to force schemaless properties to use the case of any indexes if indexes are already defined with the same name.

Finally, what would happen if you had a schema for your class and imported (via etl) some other data that had the same name but a different case than the property name -- I don't know that I exercised this but I would assume that it works by finding the correct property to shove the data into.

Either way, it's not a huge issue but it's still a bug that could cause someone to lose time (and hair).  ;)

**Comment by TifosiTech** [comment]
I take that back -- there is only 1 property called PlayerId.

Here are the results:

**REPRO**: ETL using PlayerID but data uses PlayerId

(These queries fail.)
x SELECT FROM BadPlayer WHERE PlayerId = 99
x SELECT FROM BadPlayer WHERE PlayerID = 99
x SELECT FROM BadPlayer WHERE PlayerID >= 99 AND PlayerID <= 99

(This query works but returns only 1 property called PlayerId.)
**SELECT FROM BadPlayer WHERE PlayerId >= 99 AND PlayerId <= 99**

      {
          "result": [
              {
                  "@type": "d",
                  "@rid": "#13:100",
                  "@version": 1,
                  "@class": "BadPlayer",
                  "PlayerId": 99,

**Comment by tglman** [comment]
I'm going to postpone it's not critical

**Comment by lvca** [comment]
Did you already fix this in v2.2.0?

**Comment by luigidellaquila** [comment]
Hi guys

property names are case sensitive, so the fact that the first query does not return any records is correct.
There is a known problem that makes property names case insensitive when the schema is defined, I think that's what has to be fixed.

Please, always take into consideration property name case when writing queries

Thanks

Luigi
